esx_service/utils/auth_data.py

```diff
@@ -896,19 +895,20 @@ def _remove_volumes_for_tenant(self, tenant_id):
                 dir_paths.add(vmdk['path'])
                 logging.debug("path=%s filename=%s", vmdk['path'], vmdk['filename'])
                 logging.debug("Deleting volume path%s", vmdk_path)
+                datastore_url = vmdk_utils.get_datastore_url(vmdk['datastore'])
                 err = vmdk_ops.removeVMDK(vmdk_path=vmdk_path,
                                           vol_name=vmdk_utils.strip_vmdk_extension(vmdk['filename']),
                                           vm_name=None,
                                           tenant_uuid=tenant_id,
-                                          datastore=vmdk['datastore'])
+                                          datastore_url=datastore_url)
                 if err:
                     logging.error("remove vmdk %s failed with error %s", vmdk_path, err)
                     error_msg += str(err)
 
             VOL_RM_LOG_PREFIX = "Tenant <name> %s removal: "
             # delete the symlink /vmfs/volume/datastore_name/tenant_name
             # which point to /vmfs/volumes/datastore_name/tenant_uuid
-            for (datastore, url_name, path) in vmdk_utils.get_datastores():
+            for (datastore, url_name, path, url) in vmdk_utils.get_datastores():
```

Comment:
nit - this section seems to be cut-n-pasted  afew times, better to have  a function